src/_shared/js/messages.js

```diff
@@ -16,11 +17,13 @@ export function getIframeId() {
             try {
                 json = JSON.parse(evt.data);
             } catch(_) { return; }
+
             const keys = Object.keys(json);
-            if( !(keys.length === 1 && keys[0] === 'id' )) return;
+            if( keys.length !== 2 || !keys.includes('id') || !keys.includes('host') ) return;
 
             self.removeEventListener('message', onMessage);
-            resolve(iframeId = json.id);
+            ({ id: iframeId, host: parentOrigin } = json);
```

Comment:
yeah destructuring is a godsend for you fingers 😆  I started learning standard ML and pattern matching is soooo easy to read.